extensions/cxf/runtime/src/main/java/io/quarkus/cxf/runtime/CXFQuarkusServlet.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.cxf.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletConfig;
+
+import org.apache.cxf.Bus;
+import org.apache.cxf.BusFactory;
+import org.apache.cxf.frontend.ServerFactoryBean;
+import org.apache.cxf.transport.servlet.CXFNonSpringServlet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CXFQuarkusServlet extends CXFNonSpringServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CXFQuarkusServlet.class);
+
+    public static class WebServiceConfig {
+        private String path;
+        private String className;
+
+        public WebServiceConfig(String path, String className) {
+            super();
+            this.path = path;
+            this.className = className;
+        }
+
+        public String getClassName() {
+            return className;
+        }
+
+        public String getPath() {
+            return path;
+        }
+
+        @Override
+        public String toString() {
+            return "Web Service " + className + " on " + path;
+        }
+
+    }
+
+    private static final long serialVersionUID = 1L;
+
+    private static final List<WebServiceConfig> WEB_SERVICES = new ArrayList<>();
+
+    @Override
+    public void loadBus(ServletConfig servletConfig) {
+        super.loadBus(servletConfig);
+
+        // You could add the endpoint publish codes here
+        Bus bus = getBus();
+        BusFactory.setDefaultBus(bus);
+
+        // You can als use the simple frontend API to do this
+        ServerFactoryBean factory = new ServerFactoryBean();
+        factory.setBus(bus);
+
+        for (WebServiceConfig config : WEB_SERVICES) {
+            try {
+                Class<?> serviceClass = Thread.currentThread().getContextClassLoader().loadClass(config.getClassName());
+
+                factory.setServiceClass(serviceClass);
```

Comment:
If I'm not mistaken, when using #setServiceClass(Class), the new instance is created by CXF, thus the endpoint bean is not managed by the DI container and can't rely on other beans.
I think that's the tricky part to integrate with Quarkus build-time io.quarkus.arc.runtime.BeanContainer, to do the class lookup in Arc and use #setServiceBean(Object) at runtime with the instance instead.